lib/host/wasi/inode-linux.cpp

```diff
@@ -751,6 +751,119 @@ WasiExpect<Poller> INode::pollOneoff(__wasi_size_t NSubscriptions) noexcept {
     return WasiUnexpect(__WASI_ERRNO_NOMEM);
   }
 }
+WasiExpect<void>
+INode::getAddrinfo(const char *NodeStr, const char *ServiceStr,
+                   const __wasi_addrinfo_t &Hint, uint32_t MaxResLength,
+                   std::vector<struct __wasi_addrinfo_t *> *WasiAddrinfoArray,
+                   std::vector<struct __wasi_sockaddr_t *> *WasiSockaddrArray,
+                   std::vector<char *> *AiAddrSaDataArray,
+                   std::vector<char *> *AiCanonnameArray,
+                   /*Out*/ __wasi_size_t *ResLength) noexcept {
```

Comment:
All of these variables must not be `nullptr`, so we should use reference instead.